src/vs/workbench/services/themes/common/colorThemeSchema.ts

```diff
@@ -218,44 +219,101 @@ const textmateColorSchema: IJSONSchema = {
 };
 
 export const colorThemeSchemaId = 'vscode://schemas/color-theme';
+export const workbenchColorsSchemaForThemeId = 'vscode://schemas/workbench-color-for-theme';
+export const themePaletteEnumSchemaId = 'vscode://schemas/theme-palette-enum';
+let colorThemeSchema: IJSONSchema = generateColorThemeSchemas();
 
-const colorThemeSchema: IJSONSchema = {
-	type: 'object',
-	allowComments: true,
-	allowTrailingCommas: true,
-	properties: {
-		colors: {
-			description: nls.localize('schema.workbenchColors', 'Colors in the workbench'),
-			$ref: workbenchColorsSchemaId,
-			additionalProperties: false
-		},
-		tokenColors: {
-			anyOf: [{
-				type: 'string',
-				description: nls.localize('schema.tokenColors.path', 'Path to a tmTheme file (relative to the current file).')
-			},
-			{
-				description: nls.localize('schema.colors', 'Colors for syntax highlighting'),
-				$ref: textmateColorsSchemaId
-			}
-			]
-		},
-		semanticHighlighting: {
-			type: 'boolean',
-			description: nls.localize('schema.supportsSemanticHighlighting', 'Whether semantic highlighting should be enabled for this theme.')
-		},
-		semanticTokenColors: {
-			type: 'object',
-			description: nls.localize('schema.semanticTokenColors', 'Colors for semantic tokens'),
-			$ref: tokenStylingSchemaId
+function generateColorThemeSchemas() {
+	const schemaRegistry = Registry.as<IJSONContributionRegistry>(JSONExtensions.JSONContribution);
+	const originalWorkbenchColorsSchema = schemaRegistry.getSchemaContributions().schemas[workbenchColorsSchemaId];
+	const workbenchColorSchemaForTheme = JSON.parse(JSON.stringify(originalWorkbenchColorsSchema));
+	const themePaletteEnumSchema = schemaRegistry.getSchemaContributions().schemas[themePaletteEnumSchemaId];
+
+	if (themePaletteEnumSchema?.enum !== undefined && themePaletteEnumSchema.enum.length > 0) {
+		for (const key in workbenchColorSchemaForTheme.properties) {
```

Comment:
This is the location where the workbench color schema is modified. 
If a palette is defined, then every color item (`workbenchColorSchemaForTheme.properties`) can either use the color-hex, or the var name from the palette.
If there is no palette, then the schema will exactly be the same with the original `workbenchColorsSchema`